SoftLayer/managers/hardware.py

```diff
@@ -257,6 +257,28 @@ def rescue(self, hardware_id):
         """
         return self.hardware.bootToRescueLayer(id=hardware_id)
 
+    def trunk(self, hardware_id, public, vlan_id):
+        """Add vlan trunk to a server.
+
+        :param integer hardware_id: the hardware ID to add vlan trunk
+        :param integer vlan_id: the vlan number
+
+        Example::
+
+            result = mgr.trunk(824671, 1234)
+        """
+        hw = self.get_hardware(hardware_id)
+        primary = 'primaryBackendIpAddress'
+        if public:
+            primary = 'primaryFrontendIpAddress'
+        cids = [c['id'] for c in hw['networkComponents']
```

Comment:
You can avoid doing this awkward filtering by masking in primaryNetworkComponent or primaryBackendNetworkComponent instead of using get_hardware().

E.G.:
```
hw = self.client.call('Hardware', 'getObject', mask='id,primaryNetworkComponentprimaryBackendNetworkComponent')
```